presto-main/src/main/java/com/facebook/presto/connector/informationSchema/InformationSchemaTableLayoutHandle.java

```diff
@@ -13,27 +13,29 @@
  */
 package com.facebook.presto.connector.informationSchema;
 
-import com.facebook.presto.spi.ColumnHandle;
+import com.facebook.presto.metadata.QualifiedTablePrefix;
 import com.facebook.presto.spi.ConnectorTableLayoutHandle;
-import com.facebook.presto.spi.predicate.TupleDomain;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Set;
 
 import static java.util.Objects.requireNonNull;
 
 public class InformationSchemaTableLayoutHandle
         implements ConnectorTableLayoutHandle
 {
     private final InformationSchemaTableHandle table;
-    private final TupleDomain<ColumnHandle> constraint;
+    private final Set<QualifiedTablePrefix> prefixes;
```

Comment:
Maybe this should be `Optional`. `Optional.empty` would mean full scan. `Optional.empty()` would be used if too many prefixes are selected (e.g: above 5000)

Reply:
But https://github.com/prestodb/presto/pull/10356 solves a different problem, right? Here we just wouldn't send all the `QualifiedTablePrefixex` if there is tons of them